src/InterfaceSymbolicUtils.jl

```diff
@@ -26,204 +26,204 @@ will generate a symbolic equation in SymbolicUtils.jl format.
-            end
-            out = dummy_op(left_side, right_side)
-            # Can probably delete this check TODO
-            @return_on_false isgood(out) Inf
-            return out, true
-        end
+        throw(AssertionError("Function $(String(x.name)) has ariety > 2 !"))
     end
 end
 
-# Just constant
-function symbolic_to_node(eqn::T, options::Options;
-                     varMap::Union{Array{String, 1}, Nothing}=nothing)::Node where {T<:Number}
-    return Node(convert(CONST_TYPE, eqn))
-end
-
-# Just variable
-function symbolic_to_node(eqn::T, options::Options;
-                     varMap::Union{Array{String, 1}, Nothing}=nothing)::Node where {T<:SymbolicUtils.Sym{<:Number}}
-    return Node(varMap_to_index(eqn.name, varMap))
-end
-
-function _multiarg_split(op_idx::Int, eqn::Array{Any, 1},
-                        options::Options, varMap::Union{Array{String, 1}, Nothing}
-                       )::Node
-    if length(eqn) == 2
-        return Node(op_idx,
-                    symbolic_to_node(eqn[1], options, varMap=varMap),
-                    symbolic_to_node(eqn[2], options, varMap=varMap))
-    elseif length(eqn) == 3
-        return Node(op_idx,
-                    symbolic_to_node(eqn[1], options, varMap=varMap),
-                    _multiarg_split(op_idx, eqn[2:3], options, varMap))
+# Split equation
+function split_eq(op, args, opt)
+    !(op ∈ (sum, prod, +, *)) && throw(error("Unsupported operation $op in expression!"))
+    if Symbol(op) == Symbol(sum)
+        ind = findoperation(+, opt.binops)
+    elseif Symbol(op) === Symbol(prod)
+        ind = findoperation(*, opt.binops)
     else
-        # Minimize depth:
-        split_point = round(Int, length(eqn) // 2)
-        return Node(op_idx,
-                    _multiarg_split(op_idx, eqn[1:split_point], options, varMap),
-                    _multiarg_split(op_idx, eqn[split_point+1:end], options, varMap))
+        ind = findoperation(op, opt.binops)
     end
+    return Node(ind, convert(Node, args[1], opt), convert(Node, op(args[2:end]...), opt))
 end
 
-# Equation:
-function symbolic_to_node(eqn::T, options::Options;
-                       varMap::Union{Array{String, 1}, Nothing}=nothing
-                  )::Node where {T<:SymbolicUtils.Term{<:Number}}
-    args = SymbolicUtils.arguments(eqn)
-    l = symbolic_to_node(args[1], options, varMap=varMap)
-    nargs = length(args)
-    op = SymbolicUtils.operation(eqn)
-    if nargs == 1
-        op_idx = unaop_to_index(op, options)
-        return Node(op_idx, l)
-    else
-        op_idx = binop_to_index(op, options)
-        if nargs == 2
-            r = symbolic_to_node(args[2], options, varMap=varMap)
-            return Node(op_idx, l, r)
-        else
-            # TODO: Assert operator is +, *
-            return _multiarg_split(op_idx, args, options, varMap)
-        end
+function findoperation(op, ops)
+    for (i,oi) in enumerate(ops)
+        Symbol(oi) == Symbol(op) && return i
```

Comment:
Here.

Reply:
Ah, forget it :) . I should really wait some time before waking up. We match the symbols.